syntax = "proto3";

package proto;

import "google/protobuf/timestamp.proto";

service Emu {
  rpc GetCurrentEventStatus(EventStatusRequest) returns (LegacyEventUpdate);
  rpc CreateEventStatus(CreateEventStatusReq) returns (CreateEventStatusResp);
}

message EventStatusRequest {
  uint64 event_id = 1;
}

message LegacyEventUpdate {
  uint64 event_id = 1;
  uint32 percentage = 2;
  enum ActionStatus {
    IN_PROGRESS = 0;
    DONE = 1;
    ERROR = 2;
  }
  ActionStatus action_status = 3;
  google.protobuf.Timestamp time = 4;
  string description = 5;
}

message CreateEventStatusReq {
}

message CreateEventStatusResp {
  string event_id = 2;
}
